examples/tabular/example_custom_feature_generator.py

```diff
@@ -0,0 +1,170 @@
+sample_train_data = train_data.head(100)  # subsample for faster demo
+
+# Separate features and labels
+# Make sure to not include your label/target column when sending input to the feature generators, or else the label will be transformed as well.
+X = sample_train_data.drop(columns=[label_column])
+y = sample_train_data[label_column]
+
+X_test = test_data.drop(columns=[label_column])
+y_test = test_data[label_column]
+
+print(X)
+
+##############################
+# Fitting feature generators #
+##############################
+
+from autogluon.utils.tabular.features.generators import CategoryFeatureGenerator, IdentityFeatureGenerator
+
+# IdentityFeatureGenerator is a 'do-nothing' feature generator if given default arguments. It will simply pass the data along.
+identity_feature_generator = IdentityFeatureGenerator()
+
+# fit_transform the generator using the input data. This must be done prior to calling transform.
+X_transform = identity_feature_generator.fit_transform(X=X, verbosity=3)  # verbosity=3 to log more information during fit.
+# identity_feature_generator.fit(X=X)  # This is identical to fit_transform, just without returning X_identity_out
+
+# Because IdentityFeatureGenerator simply passes the data along, nothing changed.
+assert X_transform.equals(X)
+
+identity_feature_generator = IdentityFeatureGenerator(features_in=['age', 'workclass'])  # Limit the valid input to only 'age' and 'workclass' features.
+X_transform = identity_feature_generator.fit_transform(X=X, verbosity=3)
+print(X_transform.head(5))  # Now the output only contains the two features we declared in the input arguments to the generator, acting as a feature filter.
+
+from autogluon.utils.tabular.features.feature_metadata import R_INT
+identity_feature_generator = IdentityFeatureGenerator(infer_features_in_args={'valid_raw_types': [R_INT]}, verbosity=3)  # Limit the valid input to only integer features.
+X_transform = identity_feature_generator.fit_transform(X=X)
+print(X_transform.head(5))  # Now the output only contains the int type features, acting as a type filter.
+
+# Our data contains object features at present, but this is not valid input to models, so lets convert them to category types.
+category_feature_generator = CategoryFeatureGenerator(verbosity=3)
+X_transform = category_feature_generator.fit_transform(X=X)
+print(X_transform.head(5))  # Note that the int features were automatically filtered out of this output. This is due to the defaults of CategoryFeatureGenerator which does not handle features other than objects and categories.
+
+#####################################
+# Create a custom feature generator #
+#####################################
+
+from pandas import DataFrame
+from autogluon.utils.tabular.features.generators import AbstractFeatureGenerator
+
+
+# Feature generator to add 1 to all values of integer features.
+class PlusOneFeatureGenerator(AbstractFeatureGenerator):
```

Comment:
Maybe instead this generator should add `K` to all values of integer features, to demonstrate how to pass args to feature-generator, how to maintain some state, etc.

Reply:
added